app/objects/beatmap.py

```diff
@@ -654,7 +648,13 @@ async def _update_if_available(self) -> None:
                     map_md5s_to_delete.add(old_map.md5)
                 else:
                     new_map = new_maps[old_id]
-                    if old_map.md5 != new_map["file_md5"]:
+                    new_ranked_status = RankedStatus.from_osuapi(
+                        int(new_map["approved"]),
+                    )
+                    if (
+                        old_map.md5 != new_map["file_md5"]
+                        or old_map.status != new_ranked_status
+                    ):
```

Comment:
this was breaking in the case where a qualified map was updated to ranked without a change in md5. map would remain qualified indefinitely.